### PR TITLE
Rename format to _format and add get_format/1. Closes #1132

### LINKED ISSUE #1132: Consider option to opt-out of format param
We have had our first report of the `format` param clashing with a json post body. We could either namespace this param for less likely clashing, or (my preference), have an accept's plug option to opt-out /set the format param. Something like

```elixir
plug :accepts, formats: ~w(html), param: "format"
# or 
plug :accepts, formats: ~w(html), param: :none
```

We could still have the plug default to accepting a list of formats as it does today for backwards compat. Thoughts?

### PR DESCRIPTION
**Pull request by chrismccord**
@josevalim I could use a review here when you have a moment

### CODE REVIEW

**Comment by chrismccord** [comment]
This change also leave the user unable to pattern match the format in the controller, unless they happen to pass the query param. This is easily supported though with a simple plug:

```elixir
pipeline :browser do
  ...
  plug :accepts, ~w(html json)
  plug :put_format_param
end

...
defp put_format_param(conn, _) do
  put_in conn.params, "_format", Phoenix.Controller.get_format(conn)
end
```

Is this sufficient to have users throw into their project as needed?

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,10 @@
 * Deprecations
   * [Endpoint] `render_errors: [default_format: "html"]` is deprecated in favor of `render_errors: [accepts: ["html"]]`
 
+* Backward incompatible changes
+  * [Controller] - The "format" param for overriding the accept header has been renamed to "_format" and is no longer injected into the params when parsing the Accept headers. Use `get_format/1` to access the negotiated format.
```
Comment (josevalim) [inline CHANGELOG.md]: `_format` because of markdown syntax.

File: lib/phoenix/controller.ex [hunk 2]
```diff
@@ -605,6 +605,20 @@ defmodule Phoenix.Controller do
   end
 
   @doc """
+  Puts the format in the connection.
+
+  See `get_format/1` for retrieval
```
Comment (josevalim) [inline lib/phoenix/controller.ex]: Missing `.`.

**Comment by josevalim** [comment]
:shipit:

**Comment by josevalim** [comment]
> Is this sufficient to have users throw into their project as needed?

Yes.

**Pull request merged**